Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +212,49 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
+        $document = $this->dm->find(null, $path);
 
-        return $this->dm->find(null, $path);
+        if ($document instanceof AutoRouteInterface) {
+            return $document;
+        }
+
+        return null;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * Convert the given generic document to an auto route document, migrating any children which it may have.
+     *
+     * @param Generic $document
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
```

Comment:
@dantleech @lsmith77 I think that the `flush()` call in the `migrateGenericToAutoRoute` causes the `AutoRouteListener` to be called again...

```
1) Symfony\Cmf\Bundle\RoutingAutoBundle\Tests\Functional\EventListener\AutoRouteListenerTest::testGenericNodeShouldBeConvertedInAnAutoRouteNode
Symfony\Cmf\Component\RoutingAuto\ConflictResolver\Exception\ExistingUriException: There already exists an auto route for URL "/blog" and the system is configured to throw this exception in this case. Alternatively you can choose to use a different strategy, for example, auto incrementation. Please refer to the documentation for more information.

/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony-cmf/routing-auto/ConflictResolver/ThrowExceptionConflictResolver.php:32
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony-cmf/routing-auto/UriGenerator.php:126
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony-cmf/routing-auto/AutoRouteManager.php:123
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony-cmf/routing-auto/AutoRouteManager.php:80
/Users/manuele/Work/Projects/RoutingAutoBundle/Doctrine/Phpcr/AutoRouteListener.php:68
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony/symfony/src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php:63
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/UnitOfWork.php:3944
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/UnitOfWork.php:2183
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/DocumentManager.php:752
/Users/manuele/Work/Projects/RoutingAutoBundle/Adapter/PhpcrOdmAdapter.php:245
/Users/manuele/Work/Projects/RoutingAutoBundle/Adapter/PhpcrOdmAdapter.php:151
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony-cmf/routing-auto/AutoRouteManager.php:85
/Users/manuele/Work/Projects/RoutingAutoBundle/Doctrine/Phpcr/AutoRouteListener.php:68
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/symfony/symfony/src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php:63
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/UnitOfWork.php:3944
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/UnitOfWork.php:2183
/Users/manuele/Work/Projects/RoutingAutoBundle/vendor/doctrine/phpcr-odm/lib/Doctrine/ODM/PHPCR/DocumentManager.php:752
/Users/manuele/Work/Projects/RoutingAutoBundle/Tests/Functional/EventListener/AutoRouteListenerTest.php:577
```